addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -2614,6 +2614,41 @@ export class OdooEditor extends EventTarget {
      *
      */
     initElementForEdition(element = this.editable) {
+        // Detect if the editable base element contain orphan inline nodes.
+        // If so we transform the base element HTML to put those orphans
+        // inside `<p>` containers.
+        const orphanInlineChildNodes = [...element.childNodes].filter(
+            (n) => !isBlock(n) && (n.nodeType === Node.ELEMENT_NODE || n.textContent.trim() !== "")
```

Comment:
I'm not sure this check is adequate. I suggest we discuss it tomorrow in 3D. 😎

Reply:
Summary of the discussion : 
Transform all root childNodes that are not block BUT not spaces used for formatting.